app/vep/utils/vcf_results.py

```diff
@@ -144,117 +147,131 @@ def _get_alt_allele_details(
     )
 
 
+def _get_vcf_meta(vcf_path: FilePath) -> model.VcfMetadata:
+    """Helper method to manage metainfo for a VCF file"""
+
+    meta_path = vcf_path.with_name(META_FILE)
+    if not meta_path.exists():
+        variant_count_str = subprocess.check_output(
+            f"bcftools stats {vcf_path} | grep 'number of records:'",
+            shell=True, text=True
+        )
+        header_count_str = subprocess.check_output(
+            f"bcftools view -h {vcf_path} | wc -l",
+            shell=True, text=True
+        )
+        vcf_info = model.VcfMetadata(
+            variant_count=int(variant_count_str.split(":")[-1]),
+            header_count=int(header_count_str)
+        )
+        with open(meta_path, "w") as meta_file:
+            meta_file.write(vcf_info.model_dump_json())
+    else:
+        with open(meta_path, "r") as meta_file:
+            vcf_info = model.VcfMetadata.model_validate_json(meta_file.read())
+    return vcf_info
+
+
 def get_results_from_path(
-    page_size: int, page: int, vcf_path: str
+    page_size: int, page: int, vcf_path: FilePath
 ) -> model.VepResultsResponse:
-    """Helper method that converts a file path to a stream
-    for use with get_results_from_stream"""
-    # Todo check file file exists
-    vcf_records = vcfpy.Reader.from_path(vcf_path)
-    return _get_results_from_vcfpy(page_size, page, vcf_records)
+    """Returns a page of VCF data from the given filepath.
+    Slices the input VCF file to a smaller one
+    and converts it to stream for get_results_from_stream"""
+    if not vcf_path.exists():
+        raise Exception(f"VEP results file not found at {vcf_path}")
+
+    # Fetch a pageful of variant records with headers
+    vcf_info = _get_vcf_meta(vcf_path)
+    total = vcf_info.variant_count
+    page = max(page, 1) # normalize values
+    page_size = min(max(page_size, 0), total)
+    row_offset = min(page * page_size, total) + vcf_info.header_count
+    vcf_headers = subprocess.check_output( # fetch all header rows
+        f"bcftools view -h {vcf_path}", shell=True, text=True
+    )
+    vcf_slice = subprocess.check_output( # fetch subset of variant rows
+        f"bcftools view {vcf_path} | head -n{row_offset} | tail -n{page_size}",
+        shell=True, text=True
+    )
+    vcf_stream = StringIO(vcf_headers + vcf_slice)
```

Comment:
Ideally this should be just in `_get_results_from_vcfpy`, get_results_from_path and get_results_from_stream are just public interfaces to the `_get_results_from_vcfpy` 

EDIT
Ah I see, `_get_results_from_vcfpy` works from a stream but the meta function uses sub processes that operate on files. 

The streaming is used to support embedded vcf test data in the unit tests. It might be worth rethinking the tests given that `app` doesn't resolve from pytest

Reply:
Yes, I noticed the stream consumer was only used in tests. However, the single-page vcf slice from bcftools is also a stream (no need to store) so I'll the flow is now `get_results_from_path` > `get_results_from_stream` > `_get_results_from_vcfpy`